### ISSUE TITLE
DottyDoc generation fails: `undefined ... at readTasty`

### ISSUE THREAD

**Issue by japgolly** [lampepfl/dotty#12692]
## Compiler version

3.0.0

## Reproduction

```sh
git clone https://github.com/japgolly/scalajs-react.git
cd scalajs-react
git checkout 5641c262296e2831d6fec9ff4106e91ad2dfef58
sbt doc
```

## Output

```
[warn] Setting -bootclasspath is currently not supported.
[warn] Setting -bootclasspath is currently not supported.
[warn] one warning found
[error] -- Error: core/src/main/scala-3/japgolly/scalajs/react/AsyncCallback.scala:220:6 
[error] 220 |      pa.`then`[Unit](ok, ko: js.Function1[Any, R])
[error]     |      ^^^^^^^^^^^^^^
[error]     |undefined: pa.then # -1: TermRef(AppliedType(TypeRef(TermRef(ThisType(TypeRef(NoPrefix,module class scalajs)),object js),trait Thenable),List(TypeRef(NoPrefix,type A))),then) at readTasty
[info] Main Scala API documentation successful.
[error] -- Error: core/src/main/scala/japgolly/scalajs/react/internal/JsUtil.scala:41:27 
[error] 41 |      case s: String    => js.JSON.stringify(s)
[error]    |                           ^^^^^^^^^^^^^^^^^^^
[error]    |undefined: scala.scalajs.js.JSON.stringify # -1: TermRef(TermRef(TermRef(ThisType(TypeRef(NoPrefix,module class scalajs)),object js),object JSON),stringify) at readTasty
[warn] one warning found
[error] two errors found
[error] (core / Compile / doc) DottyDoc Compilation Failed
```

**Comment by griggt** [comment]
Duplicate of #11943.

I believe it should be fixed with the combination of 3.0.1-RC1 and sbt 1.5.3.

**Comment by japgolly** [comment]
Oh whoops, sorry about that
